currency-rate/internal/broker/rate/client.go

```diff
@@ -0,0 +1,77 @@
+package rate
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log/slog"
+	"strconv"
+	"time"
+
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/broker"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/broker/transport"
+	"github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pkg/broker/transport/config"
+)
+
+var lastEventID int
+
+type Producer struct {
+	producer *transport.Producer
+}
+
+func (m *Producer) createEvent(data rateData) rateFetchedEvent {
+	lastEventID++
+	return rateFetchedEvent{
+		Event: broker.Event{
+			ID:        strconv.Itoa(lastEventID),
+			Type:      eventType,
+			Timestamp: time.Now().Format(time.RFC3339),
+		},
+		Data: data,
+	}
+}
+
+func (m *Producer) marshal(data rateFetchedEvent) ([]byte, error) {
+	bytes, err := json.Marshal(data)
+	if err != nil {
+		return nil, fmt.Errorf("marshalling command: %w", err)
+	}
+	return bytes, nil
+}
+
+func (m *Producer) SendRate(
+	ctx context.Context,
+	from string, to string, rate float32,
+) error {
+	data := m.createEvent(rateData{
+		From: from,
+		To:   to,
+		Rate: rate,
+		Time: time.Now(),
+	})
+	slog.Info(
+		"producing rate message",
+		slog.Any("from", from), slog.Any("to", to),
```

Comment:
OPTIONAL: Pavlo said in his lecture, that logging email is a bad practice, cuz emails can be considered as sensitive info